### ISSUE TITLE
Weirdness with the Holmer Terra Dos T4-40

### ISSUE THREAD

**Issue by Phobs** [Courseplay/courseplay#5023]
Could you turn the radio off for my workers, they get a little excited...
[I shake it off, I shake it off](https://i.imgur.com/jfnzuEg.gifv)

But seriously though does anyone know why it could be doing this? It's only doing this for sugar beets I checked the waypoints and there doesn't seem anything out of the ordinary, turn speed errh I meant turning circle is on (auto)13m pathfinding and turn on field are activated and the speeds are just defaults that never really give a problem. Sometimes it's just for a minute other times it's quite a bit longer.

[But I don't feel like dancin no sir no dancin today](https://i.imgur.com/Z3aqPtw.gifv)

Edit: I can't see anything wrong with the log except you guys not taking responsibility for my poor baby pandas.
Thanks :)

**Comment by Phobs** [comment]
I know it's a rather vague problem and I don't know if it's really vehicle specific or that for this vehicle the default turning circle is not big enough or if it's just this field, so whether it's reproducible is not something I can be sure about, but if someone experienced this behavior before and if they know a fix I'd appreciate it, for now I'll just try playing around with the turning circle size and see if that makes a difference.

**Comment by Tensuko** [comment]
Can't reproduce this.
In what Situation does this happen exactly, how does the Course look like ?

**Comment by Phobs** [comment]
Hi, thanks for the response, honestly I didn't think that posting a screenshot of what the course looks like mattered, I inspected the waypoints where it happened and compared them to those next to it where it didn't and they were pretty much identical. 
A simple 3 headlands going up and down on a square field of 38.500 waypoints.
I thought the gif would actually be helpful as it actually shows the shaking back and forth, explaining that sounds even weirder without some visual.. 

As you can see it goes back and forth like it can't make up it's mind how to get on the straight line and just keeps steering left and right trying to get into position to go back up and eventually it does but that can take a long time, not sure what a perfectly square box of green yellow and red blurred together lines would show you, but will take a screenshot when I have a moment to play again.

Perhaps it's worth mentioning that the course works fine on a tractor, I only ran it a bit which took long enough, it's a huge field after all, and would take like 24 hours to fully test with other machines, like I said it's just a large straight field and made sure to check a few of the spots where it happened. Which led me to believe it's the steering mode.

Which is why I asked if it's possible to change the steering mode without Courseplay forcing it back into turn steering? Which I mean.. should be possible as somewhere in a lua file? It's being forced to use it after all.. But would like to know where though, the reason I think this would solve my problem is because with all-wheel steering the front part of the machine can't move seperate from the rest at all, logically that would eliminate the front part going back and forth and instead I can just increase the turning circle.

Anyway, other then that screenshot later there really isn't much I can provide in order for you to reproduce it, I showed and explained the settings I used, checked my log for warnings or errors, gave a visual, already figured it wouldn't be reproducible on the fly like that because even I cant reproduce it when going back in the course a bit and starting it again at a close-ish waypoint where it actually happens, and sometimes it doesn't do it 30 minutes straight, other times I have to baby it, stop the course, move a bit on top of the line and start it again and it'll run for a while, or not, whatever those front wheels decide to do really.

I'll dig around in the lua files when I have a moment and will try to locate where that steering mode is forced to be used, if however you or someone else knows where that is I'd appreciate it a lot if you could point me to the right file.

**Comment by Phobs** [comment]
![fsScreen_2020_03_08_14_51_55](https://user-images.githubusercontent.com/12010927/76164185-2dfaab00-614d-11ea-9fc6-abb5ff5e623d.png)
Zoomed in or it would be a complete blur.

![fsScreen_2020_03_08_14_56_56](https://user-images.githubusercontent.com/12010927/76164195-3f43b780-614d-11ea-8a60-95b7602c4c13.png)
Here's an example of the nodes where this happens, as you can see perfectly square, it also mostly happens on this side of the field, sometimes on the other side but not as often as on this completely straight side of the field.

**Comment by Tensuko** [comment]
I hope you don't ahve the editor on all the time.
To Show the course use the 3 dots in the HUD Settings pls.

Logfile with debugchannel 9 12 and 14 on might help.

This is how the turn should look like with the debugchannels on:
![fsScreen_2020_03_08_11_51_38](https://user-images.githubusercontent.com/27344744/76164386-da895c80-614e-11ea-8cde-478bc14b086b.png)
![fsScreen_2020_03_08_11_51_48](https://user-images.githubusercontent.com/27344744/76164388-de1ce380-614e-11ea-96dc-71de2b70998b.png)

**Comment by Phobs** [comment]
Haha no no, I don't have the editor on all the time, I just turned it on to highlight the nodes where i get this behaviour I thought that would be more helpful then just that map even though it still really doesn't show anything.
Running the course now with the channels you mentioned activated, now it's just waiting till it does it again, and you'll see now that I want to show it like this it'll take forever to do it hahaha.

**Comment by Phobs** [comment]
I cut out the relevant part of the log to save you some time. This time it was unfortunately quite short but like I mentioned it varies a lot sometimes it's short sometimes it's a lot longer and sometimes it just won't move until I stop it and then move it and start it again. But looking through it these lines you can see what's happening, lastly the line it creates is the same as in your screenshot.
 
[relevant_log.txt](https://github.com/Courseplay/courseplay/files/4303351/relevant_log.txt)

**Comment by Phobs** [comment]
Alright so I looked through AITurn.lua and the comments really helped in there, I decided to try enabling vehicle.cp.settings.useAiTurns atleast I believe that's what I did by activating use AI turn maneuvers in this screenshot, I also turned on Use pathfinding in turns in that window simply because I could, I figured I'd try it out see if it worked.

And it no longer struggles with the turns at all! It just skips a little of the lane but honestly I prefer that over the struggling turns I can just get whats missed in a lane or 2 at the end no big deal. The struggling with the turns every now and then in random intervals and random lengths is way more annoying.

![fsScreen_2020_03_08_16_56_32](https://user-images.githubusercontent.com/12010927/76166441-d7966800-615e-11ea-8468-5f065621cbc5.png)
